torchqdynamics/solver_utils.py

```diff
@@ -34,19 +36,27 @@ def inv_sqrtm(mat: Tensor) -> Tensor:
     return vecs @ torch.linalg.solve(vecs, torch.diag(vals ** (-0.5)), left=False)
 
 
-def bexpect(operators: Tensor, state: Tensor) -> Tensor:
-    """Compute the expectation values of many operators on a quantum state or
-    density matrix. The method is batchable over the operators and the state.
+def bexpect(O: Tensor, x: Tensor) -> Tensor:
+    """Compute the expectation values of many operators on a state vector or a density
+    matrix.
+
+    The expectation value $\braket{O}$ of a single operator $O$ is computed
+    - as $\braket{O}=\braket{\psi|O|\psi}$ if `x` is a state vector $\psi$,
+    - as $\braket{O}=\tr(O\rho)$ if `x` is a density matrix $\rho$.
```

Comment:
A proposal, but not necessarely a good idea (not sure): use `$\braket{\hat{O}}$` instead of `$\braket{O}$`. It would be better for the documentation, but it's definitly less readable for the docstrings.

Reply:
Hahaha I _hate_ the hats on top of operators, but that's a long discussion 😅 Let's discuss this next meeting 😈